src/hmacKey.ts

```diff
@@ -0,0 +1,355 @@
+       *       // The request was an error.
+       *       console.error(err);
+       *       return;
+       *     }
+       *     hmacKey.delete((err) => {
+       *       if (err) {
+       *         console.error(err);
+       *         return;
+       *       }
+       *       // The HMAC key is deleted.
+       *     })
+       *   });
+       *
+       * //-
+       * // If the callback is omitted, a promise is returned.
+       * //-
+       * const hmacKey = storage.hmacKey('ACCESS_ID');
+       * hmacKey
+       *   .update({state: 'INACTIVE'})
+       *   .then(() => {
+       *     return hmacKey.delete();
+       *   });
+       */
+      delete: true,
+    };
+
+    super({
+      parent: storage,
+      baseUrl: `/projects/${storage.projectId}/hmacKeys`,
+      id: accessId,
+      methods,
+    });
+
+    this.accessId = accessId;
+    this.metadata = {accessId};
+    this.parent = storage;
+  }
+
+  get(options?: GetHmacKeyOptions): Promise<HmacKeyMetadataResponse>;
+  get(callback: HmacKeyMetadataCallback): void;
+  get(options: GetHmacKeyOptions, callback: HmacKeyMetadataCallback): void;
+  /**
+   * @typedef {object} GetHmacKeyOptions
+   * @property {string} userProject This parameter is currently ignored.
+   */
+  /**
+   * Retrieves and populate an HMAC key's metadata.
+   *
+   * HmacKey.get() does not give the HMAC key secret, as
+   * it is only returned on creation.
+   * The authenticated user must have `storage.hmacKeys.get` permission for the project in which the key exists.
+   *
+   * @param {GetHmacKeyOptions} [options] Configuration options.
+   * @param {HmacKeyMetadataCallback} [callback] Callback function.
+   * @returns {Promise<HmacKeyMetadataResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   *
+   * //-
+   * // Get the HmacKey's Metadata.
+   * //-
+   * storage.hmacKey('ACCESS_ID')
+   *   .get((err, hmacKeyMetadata) => {
+   *     if (err) {
+   *       // The request was an error.
+   *       console.error(err);
+   *       return;
+   *     }
+   *     console.log(hmacKeyMetadata);
+   *   });
+   *
+   * //-
+   * // If the callback is omitted, a promise is returned.
+   * //-
+   * storage.hmacKey('ACCESS_ID')
+   *   .get()
+   *   .then((data) => {
+   *     const hmacKeyMetadata = data[0];
+   *     console.log(hmacKeyMetadata);
+   *   });
+   */
+  get(
+    optionsOrCb?: GetHmacKeyOptions | HmacKeyMetadataCallback,
+    cb?: HmacKeyMetadataCallback
+  ): Promise<HmacKeyMetadataResponse> | void {
+    const {options, callback} = normalize<
+      GetHmacKeyOptions,
+      HmacKeyMetadataCallback
+    >(optionsOrCb, cb);
+    const opts = Object.assign({}, options);
+    // autoCreate is ignored - key must be created using Storage.createHmacKey.
+    delete opts.autoCreate;
+
+    const reqOpts = {
+      uri: '/',
+      qs: opts,
+    };
+
+    this.request(reqOpts, (err, metadata, res) => {
```

Comment:
Did you try the default behavior of `get` here, that is set by the ServiceObject class?

Reply:
Ah I see. `ServiceObject.get()` wraps `getMetadata()` and auto-creates the object if it doesn't exist.

Auto-create is not defined for HMAC key, so we should just expose the `getMetadata()` method inherited from `ServiceObject`. Does that make sense? @stephenplusplus

Reply:
We can actually keep "get()", and if that autoCreate setting is passed, it would be ignored if we don't have a "create" method defined. But just curious, why can't we have the create method? If it is because it requires additional configuration, that is actually possible to be defined in the get(options) argument.

Reply:
The `id` (accessId) is generated by the server, so `autoCreate` wouldn't make sense for an HMAC key.

Reply:
I should explain more: an HMAC key is different from, say, a Bucket where a user defines the name (id) of the possibly non-existent resource.

Reply:
Ah, gotcha. Thanks for explaining! So I think you can still pass 'methods.get = true' in the constructor and as long as methods.create isn't set, it'll work as expected. Let me know if I'm wrong on that :)

Reply:
Yup, looks like that works. On second thought, I think we should expose `getMetadata` rather than `get`, since the get method doesn't make much sense for HMAC keys without `autoCreate`.

Reply:
`get`'s purpose has rightly been [called into question before](https://github.com/googleapis/nodejs-common/issues/234), but it is still something we expose across the libraries wherever `getMetadata` exists, even if `autoCreate` isn't available. I would vote to keep that going for the sake of parity.

Reply:
Added `get` and `getMetadata` methods (inherited from `ServiceObject`)